### ISSUE TITLE
Playwright: support clearing cookies in mid-test

### ISSUE THREAD

**Issue by worldomonation** [Automattic/wp-calypso#53903]
**Task Details**
Currently, clearing cookies anywhere within a top-level `describe` would cause subsequent tests to fail.

For example, take this structure:

```
describe( top suite )
      describe( test case 1)
             test steps...
             
      describe( test case 2)
             clear cookie
             test steps...

      describe( test case 3)
             test steps...
```

Test steps in `test case 2` and the whole of `test case 3` would end up failing because the cookies have been cleared.

I am not fully certain why this is the case, but it might have to do with the reuse of the same `BrowserContext` for the entire top suite is the problem.

**Related to**
<-- If there are any other GitHub issue(s) that readers should be aware of, include them here. -->

**Comment by worldomonation** [comment]
This issue was worked around somewhat in #53931 by creating a new post, closing the current context then starting a new context.

It would seem that even after clearing the cookies and confirming the cookies are empty, that WPCOM has some 'memory' of the fact that user was logged in. This results in the following sequence of events not working as expected when it should:

1. log in
2. create and publish post
3. log out
4. clear cookie
5. visit the published post in Step 2
6. click `like`

At this point, the expectation is that a popup window appears with the Login form.
However, what actually ends up happening is the Login popup disappears immediately, aborting the `logInFromPopup` method.

**Comment by worldomonation** [comment]
This is fixed in #54740 in `browser-manager`.
